### PR TITLE
refactor/docs: Add Showcase & CodeSnippet components w/ import.meta.glob pattern for .mdx routes

### PR DESCRIPTION
**Pull request by maiieul**
# What is it?

- [ ] Feature / enhancement
- [ ] Bug
- [ ] Docs / tests

# Description

Best way to test this is probably to manually check every route. I did so in `dev`, `preview` and `preview.cloudflare`.

- Adds `Showcase` and `CodeSnippet` components that take advantage of `import.meta.glob`. 
   - In order for the `CodeSnippet` component to automatically import the snippets, I put all the snippets into a _/snippets_ subfolders. For example, if we are at _/docs/headless/modal_, the snippets will be in _/docs/headless/modal/snippets_.
   -  In order for the `Showcase` component to automatically import the the component examples and their raw code value, I put all the example components into _/examples_ subfolders. For example, if we are at _/docs/headless/modal_, the example components will be in _/docs/headless/modal/examples_.

> So for `Showcase` and `CodeSnippet` to work we will have to remember to put the example components into the route's _/examples_ subfolder, and the snippets into the route's _/snippets_ subfolder.

Except for autocomplete (I remember this is supposed to be removed, right?) I rewrote all the routes to take advantage of this new pattern that saves us the hassle of having an examples.tsx and exports.ts file to define all the components, export them, and then import them in the .mdx file.

### - Moved components from _/headless/(components)_ into _/headless_ and _/fluffy/(components)_ into _/fluffy_ 

**Explanation:**

TLDR; when using `import.meta.glob`, it's best to **NOT** use grouped layouts as it makes working with the `useLocation()` hook (`location.url.pathname`) harder.

[Grouped layouts](https://qwik.builder.io/docs/advanced/routing/#grouped-layouts) are particularly useful for organizing the codebase logically.

But this PR adds the `import.meta.glob` pattern with a wildcard search for retrieving the files.

Ex:

```tsx
const components: any = import.meta.glob('/src/routes/docs/**/**/examples/*.tsx', {
  import: 'default',
  eager: isDev ? false : true,
});
```

Will allow me to retrieve 

`/src/routes/docs/components/headless/modal/examples/hero.tsx` when the URL is at `/docs/headless/modal`
Or `/src/routes/docs/components/fluffy/modal/examples/hero.tsx` when the URL is at `/docs/fluffy/modal`.

In order to do so, I need to use the `useLocation()` hook in the component like that:

```tsx
  const location = useLocation();
  const componentPath = `/src/routes${location.url.pathname}examples/hero.tsx`;
```

This code can be used **only** when there isn't a grouped layout (here the `/(components)` folder I removed). This is simpler and more resilient to code changes.

If we use grouped layouts, then we can't directly use `location.url.pathname`. We have to change the URL string to add `"/(components)"` in it, and we have to change the replace function every time we move files around.

# Checklist:

- [x] My code follows the [developer guidelines of this project](https://github.com/qwikifiers/qwik-ui/blob/main/CONTRIBUTING.md)
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] Added new tests to cover the fix / functionality

### CODE REVIEW

**Comment by thejackshelton** [comment]
Looks good! 

Comments about this PR here:
https://discord.com/channels/990511757091033108/1172225319009275934/1179118851586281612
